target/linux/mpc85xx/files/arch/powerpc/boot/dts/ecw7210.dts

```diff
@@ -0,0 +1,304 @@
+/*
+ * Edge-core ECW-7210-L Device Tree Source
+ *
+ * Copyright (C) 2019 Scott Gilliland <zeroping@gmail.com>
+ *
+ * This program is free software; you can redistribute  it and/or modify it
+ * under  the terms of  the GNU General  Public License as published by the
+ * Free Software Foundation;  either version 2 of the  License, or (at your
+ * option) any later version.
+ */
+
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/include/ "fsl/p1020si-pre.dtsi"
+/ {
+	model = "Edge-Core ECW7210-L";
+	compatible = "edgecore,ecw7210";
+
+	memory {
+		device_type = "memory";
+	};
+
+	board_lbc: lbc: localbus@ffe05000 {
+		reg = <0 0xffe05000 0 0x1000>;
+		ranges = <0x0 0x0 0x0 0xee000000 0x2000000
+			  0x1 0x0 0x0 0xffa00000 0x0040000>;
+		interrupts = <0x13 0x2>;
+		#size-cells = <0x1>;
+		#address-cells = <0x2>;
+		bus-frequency = <24999999>;
+		nand@1,0 {
+			#address-cells = <0x1>;
+			#size-cells = <0x1>;
+			compatible = "fsl,p1020-fcm-nand", "fsl,elbc-fcm-nand";
+			reg = <0x1 0x0 0x40000>;
+			partition@0 {
+				reg = <0x0 0x1000000>;
+				label = "kernel";
+			};
+			partition@01000000 {
+				reg = <0x1000000 0x1f000000>;
+				label = "ubi";
+			};
+			
+			//label = "ubi";
+			//compatible = "ubi,volume";
+		};
+		
+		// may be best to leave the NOR flash alone, since we can put everything in NAND.
+		nor@0,0 {
+			device-width = <0x1>;
+			bank-width = <0x2>;
+			reg = <0x0 0x0 0x2000000>;
+			compatible = "cfi-flash";
+			#size-cells = <0x1>;
+			#address-cells = <0x1>;
+
+			partition@1f80000 {
+				read-only;
+				label = "mtd-uboot-image";
+				reg = <0x1f80000 0x80000>;
+			};
+
+			partition@1f60000 {
+				label = "uboot-env";
+				reg = <0x1f60000 0x20000>;
+			};
+
+			partition@1f40000 {
+				read-only;
+				label = "u-env-backup";
+				reg = <0x1f40000 0x20000>;
+			};
+
+			partition@1f20000 {
+				read-only;
+				label = "mtd-mfr";
+				reg = <0x1f20000 0x20000>;
+			};
+
+			partition@1e20000 {
+				read-only;
+				label = "mtd-reserved";
+				reg = <0x1e20000 0x100000>;
+			};
+
+			partition@1e00000 {
+				read-only;
+				label = "crashlog";
+				reg = <0x1e00000 0x20000>;
+			};
+
+			partition@60000 {
+				read-only;
+				label = "mtd-unused";
+				reg = <0x60000 0x1da0000>;
+			};
+
+			partition@20000 {
+				read-only;
+				label = "mtd-dtb";
+				reg = <0x20000 0x40000>;
+			};
+
+			partition@0 {
+				read-only;
+				label = "mtd-u-script";
+				reg = <0x0 0x20000>;
+			};
+		};
+	};
+
+	board_soc: soc: soc@ffe00000 {
```

Comment:
Decide for one label, if you need it. I do not think you use any of those labels.

Reply:
@adrianschmutzler due to the way the upstream tree works, you need at least the `soc` label (same goes for the localbus `lbc` label).

The `board_lbc` as well as the `board_soc` here can be dropped, as they are not used (anymore?).

Reply:
Yep, this is on my to-do list under "fix everything in the .dts because it's awful right now". I carried some cruft over from the stock device tree while fixing PCIe IRQ's. I don't fully understand how I managed to get it working, but that's not a good excuse to leave the .dts untidy.

Reply:
I've dropped the unnecessary ones. Only leaving `lbc`, `soc`, `enetx` and `pcix` to work with the included p1020 dtsi's, as blocktrron mentioned.